language: python
python: '3.6.7'
os: rhel
sudo: required
group: bluezone
addons:
  ssh_known_hosts: 9.5.161.179

before_install:
  - sudo apt-get update -y
  - sudo apt-get install -y sshpass 
  - python -m pip install --upgrade pip
  - pip install ansible six==1.12.0 junit-xml
  - pip install -r /home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/ansible_test/_data/requirements/sanity.txt
  - ansible --version

install: 
  - ansible-galaxy collection build .
  - ansible-galaxy collection install *.gz
  - ansible-playbook tests/ansible_test/_data/create_inventory.yml -e "ibmi_ip=$test_system_ip ibmi_user=$username ibmi_pass=$password ibmi_python=$python_interpreter"
  - cp -r tests/ansible_test/* /home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/ansible_test/
  - cat /home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/ansible_test/_data/inventory

script:
  - pushd ~/.ansible/collections/ansible_collections/ibm/power_ibmi
  - ansible-test sanity --python 3.6 plugins/*/*.py
  #- ansible-test sanity --python 3.6 plugins/*/*/*.py
  #- ansible-test integration --python 3.6 -v --allow-unsupported --continue-on-error $(find tests/integration/targets -name "ibmi*" -type d -printf "%P ")
  #- ansible-test integration --python 3.6 -v --start-at ibmi_lib_restore $(find tests/integration/targets -name "ibmi*" -type d -printf "%P ")
  #- ansible-test integration --python 3.6 ibmi_at
  - ansible-test integration --python 3.6 -v --continue-on-error $(find tests/integration/targets -name "ibmi*" -type d -printf "%P ")


notifications:
  #slack:
    #on_failure: always
    #rooms:
      #- ibmi-cloud
